### DISCUSSION TITLE
Create .csv files from drawing

### DISCUSSION THREAD

**Discussion by tobiokanobi** [byuflowlab/FLOWUnsteady#55] [Q&A] [answered]
Thanks for this fabulous tool. 

May I ask if there is a possibility to convert a wire/mesh/cad file of an airfoil (e.g. rotor blade) to gain the required (....) .csv files? 

Thanks,

Tobi

**Comment by EdoAlvarezR** [comment, marked as answer]
Thanks Tobi!

In the past I've exported the CAD as an STL and opened the .stl in ParaView to generate slices ([Slice filter](https://docs.paraview.org/en/latest/UsersGuide/filteringData.html#slice)) which then can be exported as CSVs from the tabular view.

However, if you already have a CAD of the blade, most likely the input to the CAD was a stack of airfoil profiles that generate the loft, so I'd recommend using the airfoil profile directly to generate the [FLOWUnsteady rotor](https://flow.byu.edu/FLOWUnsteady/api/flowunsteady-vehicle-components/#Rotor).

**Reply by tobiokanobi** [reply]
Perfect Edo! We have a coaxial setup with two different blades, each blade consinsts out of different profiles. It will be interesting how that works out.

Mixing topics a bit, but can the inertia mass be set? Our blades are on a delta-3 hinge mounted, else the upper and lower blade can flap. 

Greets,
Tobias

**Reply by EdoAlvarezR** [reply]
Can you clarify what you mean by setting "inertia mass"?

**Reply by tobiokanobi** [reply]
Sorry Eduardo for the late reply. Just recently had time to start working with FLOWUnsteady. The thing about inertia was related to rotor blades which are semi-articulated or fully articulated, hence, depending on rpm, tilt angle....the whole thing moves a bit around by itself.  Again, nice work - would love to see that it continous (maybe the propeller generation could be simplified ;-) ) .

**Reply by jayjiejie** [reply]
Hi Edo

I have a propeller blade CAD file and slice the blade into several parallel sections that are perpendicular to the propeller reference line (which is along the vertical direction). What is the best way to re-create the geometry in FLOWUnsteady using loft?

The propeller tutorial generates the geometry using propeller design parameters such as requires propeller chord distribution, pitch distribution, leading edge sweep, leading edge height distribution, aerofoil distribution, which I don't have.

Thank you very much.

Kind regards,
Shuang

**Reply by EdoAlvarezR** [reply]
Hi Shuang!

You can extract the required distributions (chord, pitch, LE, and airfoil) directly from the geometry that you already have---just process the slices to calculate the chord, pitch, and position of the LE from each section.

**Reply by jayjiejie** [reply]
Hi Edo

Thank you for your quick response. I am actually trying to create an INSEAN E779a propeller and found the E779a airfoil and station csv files in the prepopulated database (https://github.com/byuflowlab/FLOWUnsteady/tree/master/database). Is it possible to create its geometry using the files from the database?

Couple more questions for extracting the required distributions from the geometry I have (E779a) by going through the Generating Components tutorial (https://flow.byu.edu/FLOWUnsteady/api/flowunsteady-vehicle-components/):

1. With regards to the sweepdist, y/R is the y distance of LE from the middle point of hub. What is the definition of the middle point of hub? Is it the center of the hub?

2.  With regards to the heightdist,z/R is the height of leading edge from top face of hub. What is the definition of the top face of hub?

3. If I don't have the airfoil polar, can this always be generated using the xfoil = true option?

Since the marine propeller is more skewed comparing to the rotor, I am a bit unclear about the above definitions.

Kind regards,
Shuang
